network.go

```diff
@@ -944,7 +945,10 @@ func (n *network) ipamAllocateVersion(ipVer int, ipam ipamapi.Ipam) error {
 		// irrespective of whether ipam driver returned a gateway already.
 		// If none of the above is true, libnetwork will allocate one.
 		if cfg.Gateway != "" || d.Gateway == nil {
-			if d.Gateway, _, err = ipam.RequestAddress(d.PoolID, net.ParseIP(cfg.Gateway), nil); err != nil {
+			var gatewayOpts = map[string]string{
+				netlabel.Gateway: "",
```

Comment:
I would personally avoid using an empty string as a way to replace a `bool=true`. This is not readable.
Shall we use a more explicit Key and its value can be netlabel.Gateway ?